api/v1alpha1/instancegroup_types.go

```diff
@@ -187,26 +187,28 @@ type EKSSpec struct {
-	SpotPrice                   string                    `json:"spotPrice,omitempty"`
-	Tags                        []map[string]string       `json:"tags,omitempty"`
-	Labels                      map[string]string         `json:"labels,omitempty"`
-	Taints                      []corev1.Taint            `json:"taints,omitempty"`
-	UserData                    []UserDataStage           `json:"userData,omitempty"`
-	ExistingRoleName            string                    `json:"roleName,omitempty"`
-	ExistingInstanceProfileName string                    `json:"instanceProfileName,omitempty"`
-	ManagedPolicies             []string                  `json:"managedPolicies,omitempty"`
-	MetricsCollection           []string                  `json:"metricsCollection,omitempty"`
-	LifecycleHooks              []LifecycleHookSpec       `json:"lifecycleHooks,omitempty"`
-	MixedInstancesPolicy        *MixedInstancesPolicySpec `json:"mixedInstancesPolicy,omitempty"`
+	EksClusterName              string                       `json:"clusterName,omitempty"`
+	KeyPairName                 string                       `json:"keyPairName,omitempty"`
+	Image                       string                       `json:"image,omitempty"`
+	InstanceType                string                       `json:"instanceType,omitempty"`
+	NodeSecurityGroups          []string                     `json:"securityGroups,omitempty"`
+	Volumes                     []NodeVolume                 `json:"volumes,omitempty"`
+	Subnets                     []string                     `json:"subnets,omitempty"`
+	SuspendedProcesses          []string                     `json:"suspendProcesses,omitempty"`
+	BootstrapArguments          string                       `json:"bootstrapArguments,omitempty"`
+	SpotPrice                   string                       `json:"spotPrice,omitempty"`
+	Tags                        []map[string]string          `json:"tags,omitempty"`
+	Labels                      map[string]string            `json:"labels,omitempty"`
+	Taints                      []corev1.Taint               `json:"taints,omitempty"`
+	UserData                    []UserDataStage              `json:"userData,omitempty"`
+	ExistingRoleName            string                       `json:"roleName,omitempty"`
+	ExistingInstanceProfileName string                       `json:"instanceProfileName,omitempty"`
+	ManagedPolicies             []string                     `json:"managedPolicies,omitempty"`
+	MetricsCollection           []string                     `json:"metricsCollection,omitempty"`
+	LifecycleHooks              []LifecycleHookSpec          `json:"lifecycleHooks,omitempty"`
+	MixedInstancesPolicy        *MixedInstancesPolicySpec    `json:"mixedInstancesPolicy,omitempty"`
+	LicenseSpecifications       []string                     `json:"licenseSpecifications,omitempty"`
+	Placement                   *LaunchTemplatePlacementSpec `json:"placement,omitempty"`
```

Comment:
If `Placement` can be supported by LaunchConfiguration as well, we should use a more generic name e.g. `ScalingConfigurationPlacementSpec`

Reply:
Unfortunately `Placement` isn't supported by LaunchConfiguration.

Reply:
Looks like it is supported according to:
https://docs.aws.amazon.com/sdk-for-go/api/service/autoscaling/#CreateAutoScalingGroupInput

Should we have a more generic implementation in this case?

Reply:
ahh, `PlacementTenancy` is a subset of `LaunchTemplatePlacement` it's essentially the equivalent of `LaunchTemplatePlacement.tenancy`.

Are there use cases to support placement tenancy in LaunchConfigurations?

Reply:
Not currently, but in case there will be in the future - just wondering if we should have a more generic naming for the API, e.g. instead of `LaunchTemplatePlacementSpec`, just use `PlacementSpec`.
We can possibly make `.tenancy` support for launch configurations now, or down the road

Reply:
given the `PlacementSpec` has extra configurable fields that are only appropriate for LaunchTemplates, what would support for LT's and LC's both look like?

My read of your suggestion is that we would rename `LaunchTemplatePlacementSpec` -> `PlacementSpec`, and add validation that when `type == LaunchConfiguration` fail validation for any extra fields. is that what you're thinking?

Reply:
I've wired up the `PlacementTenancy` for `LaunchConfiguration`s but having some trouble validating the `EKSConfiguration` to make provide meaningful feedback depending on LC vs LT - do you have a suggestion on how we should validate this?